force-app/main/default/classes/main/standard-soql/SOQL.cls

```diff
@@ -1737,6 +1734,7 @@ public virtual inherited sharing class SOQL implements Queryable {
         }
 
         public InnerJoin with(SObjectField field) {
+            this.builder.fields.clearAllFields();
```

Comment:
does this prevent chaining multiple `with` statements to "extend" a query? I mean like 
```
// base selector
with(DEAFULT_FIELDS);
..
// extending class
MySelector.with(MORE_FIELDS);
```

Reply:
ah, ok, so top level query we can still call `with` multiple times, right?